sourcegraph/sourcegraph
# fix(cody): filter out deprecated models

Exclude deprecated models from the available models in the models config. Ensure that Cody Enterprise users have the same (similar) models varienty as the Cody Pro users. More context in [this](https://sourcegraph.slack.com/archives/C07CVL78MDF/p1723127669863409?thread_ts=1723061732.742949&cid=C07CVL78MDF) thread.

| Before | After | Dotcom |
| -- | -- | -- |
|<img width="509" alt="before" src="https://github.com/user-attachments/assets/ec71da82-139a-485d-b969-f9aa17fca264">|<img width="509" alt="after" src="https://github.com/user-attachments/assets/c5d69fa6-2e8a-4110-9fd3-e8fefda941ea">|<img width="509" alt="dotcom" src="https://github.com/user-attachments/assets/cd4eb1a5-d0d7-44c3-bdfb-0b7c9b744506">|

## Test plan

- CI
- Added unit test
- Tested locally
   - Configure Cody via "modelConfiguration" field in the site config; minimal configuration (`"modelConfiguration": {
  "sourcegraph": {}
}`) should be enough.
   - `sg start`
   - in Cody for VS Code
      - connect sing in to the local instance
      - open Cody panel
      - open model dropdown in the chat input
      - ensure that [deprecated models](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@43e06cef0cc7110e677145f7c9aae5767effdd25/-/blob/cmd/cody-gateway-config/dotcom_models.go?L118:25-118:46) are not present in the model list
